ansys/mapdl/core/post.py

```diff
@@ -519,334 +521,239 @@ def plot_element_values(
             show_elem_numbering=show_elem_numbering, **kwargs
         )
 
-    def element_displacement(
-            self, component="ALL", option="AVG"
-    ) -> np.ndarray:
-        """Return element displacement.
-
-        One value per element.  Either minimum, maximum, or average of
-        all nodes in each element.
-
-        Equilvanent MAPDL commands:
-
-        * ``ETABLE,VALUES,U,X``
-        * ``PRETAB,VALUES`` or ``*VGET,TMP,ELEM,1,ETAB,VALUES``
-
-        Parameters
-        ----------
-        component : str, optional
-            Structural displacement component to retrieve.  Must be
-            ``"X"``, ``"Y"``, ``"Z"``, ``"ALL"`` or ``"NORM"``.  Defaults to
-            ``"ALL"``.
-        option : str, optional
-            Option for storing element table data.  One of the
-            following:
-
-            * ``"MIN"`` : Store minimum element nodal value of the
-              specified item component.
-            * ``"MAX"`` : Store maximum element nodal value of the
-              specified item component.
-            * ``"AVG"`` : Store averaged element centroid value of the
-              specified item component (default).
+    def _plot_point_scalars(self, scalars, show_node_numbering=False, **kwargs):
+        """Plot point scalars"""
+        if not scalars.size:
+            raise RuntimeError(
+                "Result unavailable.  Either the result has not been loaded "
+                "with ``mapdl.set(step, sub_step)`` or the result does not "
+                "exist within the result file."
+            )
 
-        Examples
-        --------
-        Return the average element displacement for all components.
+        sel = self.selected_nodes
+        all_scalars = np.zeros(sel.shape)
+        all_scalars[sel] = scalars
 
-        >>> arr = mapdl.post_processing.element_displacement("ALL")
-        >>> arr.shape
-        (2080, 3)
-        >>> arr
-        array([[ 1.07396154e-06, -9.03608033e-06, -5.17768108e-12],
-               [ 3.15631730e-06, -2.65527340e-05,  1.07714512e-11],
-               [ 5.12543515e-06, -4.31175194e-05,  2.19929719e-12],
-               ...,
-               [ 5.41204700e-06, -4.80335719e-05,  7.75819589e-11],
-               [ 3.33649806e-06, -2.96109417e-05,  1.44947535e-10],
-               [ 1.13836132e-06, -1.01038096e-05,  6.95566641e-11]])
+        surf = self._mapdl.mesh._surf
 
-        """
-        check_elem_option(option)
-        component = component.upper()
-        check_comp(component, DISP_TYPE)
+        nods = surf["ansys_node_num"]-1
+        all_scalars = all_scalars[nods.astype(int)]
```

Comment:
This will fail if the node numbers are discontinuous.  Oftentimes, either with `numoff` or `numstr`, you'll end up with discontinuous node numbering, for example `[1, 2, 8, 9]`.  Here, this will evaluate to `[0, 1, 7, 8]`, and the highest index, `8`, will be outside the bounds of the array, `4`. I will add a test to demonstrate this issue.

Reply:
I take it back.

I put in an offset into the contact test and it worked. Crazy enough, MAPDL actually streams over an array sized out to the highest node number + 1, which is really inefficient but it gets the job done.

See my most recent commit for some minor efficiency improvements, especially using `np.empty` rather than `np.zeros`:

Sample timings:
```py
In [4]: timeit np.zeros(int(1E6))
183 µs ± 821 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [5]: timeit np.empty(int(1E6))
460 ns ± 2.55 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
Exactly! I found very weird that MAPDL stream an array of that size. But I guess it is what it is.

Thank you for the `np.empty`.